### PR TITLE
Feat nesting selectors

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**
Features:

- Support parsing nested selectors https://www.w3.org/TR/css-nesting-1/
- Fix parsing of `@document`
- Use SimpleBlock for `@media` and `@supports`, `@layer`

Notes:
- Added `TODO` - we should validate `&` in selector list in some cases, but it is still parsable

**BREAKING CHANGE:**

Yes

**Related issue (if exists):**

No

### CODE REVIEW

**Comment by kdy1** [comment]
I'm not sure why CI fails with
```

error /home/runner/work/swc/swc/node_modules/csstree: Command failed.
```

File: package.json [hunk 1]
```diff
@@ -84,6 +82,8 @@
         "core-js": "^2.6.11",
         "cross-env": "^7.0.3",
         "cspell": "^5.12.3",
+        "css-tree": "^2.0.4",
```
[the remark this answers is not in the document]
Comment (alexander-akait) [reply package.json]: Oh, it was just local test :smile: I will remove

**Comment by alexander-akait** [comment]
@kdy1 Ready for review

**Review by kdy1 — CHANGES_REQUESTED** [review]
Can you revert yarn.lock?

**Comment by alexander-akait** [comment]
@kdy1 Done

**Review by kdy1 — APPROVED** [review]

**Pull request merged**